### PR TITLE
FIXES 10043 STAGING Scenario

### PR DESCRIPTION
**Pull request by spuppala-rythmos**
by enabling the option converts templates to multiliner

### CODE REVIEW

File: src/newlines.cpp [hunk 1]
```diff
@@ -3348,6 +3348,34 @@ void newlines_cleanup_braces(bool first)
                   newline_iarf(pc, cpd.settings[UO_nl_template_class].a);
                }
             }
+            //Fixes 10043 regression
+            if (cpd.settings[UO_nl_template].b)
+            {
+               if (chunk_is_opening_brace(chunk_get_next_ncnl(pc)) || chunk_is_paren_open(chunk_get_next_ncnl(pc)))
```
Comment (mihaipopescu) [inline src/newlines.cpp]: You can cache the token

File: src/options.cpp [hunk 2]
```diff
@@ -1086,6 +1086,8 @@ void register_options(void)
                   "Newline between namespace and {.");
    unc_add_option("nl_template_class", UO_nl_template_class, AT_IARF,
                   "Add or remove newline between 'template<>' and whatever follows.");
+   unc_add_option("nl_template", UO_nl_template, AT_BOOL,
+                  "Add  newline between '>()' or '>' and '{' .");
```
Comment (mihaipopescu) [inline src/options.cpp]: Remove one space after add

File: src/options.h [hunk 3]
```diff
@@ -510,6 +510,7 @@ enum uncrustify_options
                                       // Overrides nl_after_case
    UO_nl_namespace_brace,             // newline between namespace name and brace
    UO_nl_template_class,              // newline between '>' and class in 'template <x> class'
+   UO_nl_template,                    //Add  newline between '>()' or '>' and '{' .
```
Comment (mihaipopescu) [inline src/options.h]: Add space before add. Strange that uncrustify doesn't enforce that.

File: src/options.cpp (hunk 2 above)
Comment (mihaipopescu) [reply src/options.cpp]: I'd also make the name a bit less generic.

**Review by mihaipopescu — CHANGES_REQUESTED** [review]
Looks good. Small changes.

**Review by hschlichter — APPROVED** [review]
Looks good

**Comment by hschlichter** [comment]
@mihaipopescu does fixes you asked for look good?

**Comment by spuppala-rythmos** [comment]
@hschlichter 
All changes requested by @mihaipopescu are addressed.

**Review by mihaipopescu — APPROVED** [review]

**Pull request merged**